index.js

```diff
@@ -69,6 +74,8 @@ function extendedFindQuery(model, models, { rejectUnknownProperties = false, pre
             } catch (err) {
                 if (err instanceof UnknownPropertyError) {
                     err.status = 400;
+                } else if (err instanceof UnsupportedDatasourceError){
+                    err.status = 400;
```

Comment:
Id rather return a 501 than a 400 error

Reply:
Hmm it might be both ... if one enables filtering on a datasource that is not supported then it surely a 501, if the client requests filtering on a related entity that is attached to an unsupported ds it is more like 4XX ... Tbh the fix I did is not too elegant...